langroid/agent/special/doc_chat_agent.py

```diff
@@ -860,6 +879,60 @@ def llm_hypothetical_answer(self, query: str) -> str:
                 ).content
         return answer
 
+    def add_hypothetical_questions(self, docs: List[Document]) -> List[Document]:
+        """Add hypothetical questions to documents using batch processing.
+
+        Args:
+            docs: List of documents to generate questions for
+
+        Returns:
+            List[Document]: Documents augmented with generated questions
+        """
+        if self.llm is None:
+            raise ValueError("LLM not set")
+
+        if not self.config.hypothetical_questions:
+            return docs
+
+        with status("[cyan]Generating hypothetical questions for chunks..."):
+            # Process chunks in parallel using run_batch_agent_method
+            questions_batch = run_batch_agent_method(
```

Comment:
there could be 1000+ chunks so we should use a batched version of this method. I know there's no `batch_size` param in this method, so we'd have to add that to `run_agent_method_batch` and related onces